.github/workflows/test.yml

```diff
@@ -0,0 +1,95 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: Test
+
+on:
+  push:
+    branches:
+      - '**'
+      - '!dependabot/**'
+    tags:
+      - '**'
+  pull_request:
+
+concurrency:
+  group: ${{ github.repository }}-${{ github.head_ref || github.sha }}-${{ github.workflow }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+
+env:
+  ICEBERG_HOME: /tmp/iceberg
+
+jobs:
+  ubuntu:
+    name: AMD64 Ubuntu latest
+    runs-on: ubuntu-latest
```

Comment:
Do we want to pin this to a specific version to avoid breaking the build when `latest` gets updated?

Reply:
I am in two minds with this one but I think I prefer to have `latest` so if a new OS (GitHub runner) is released we find early that there are issues with it.
The other approach might leave us with silent failures on newer systems if we don't update them periodically but both points are valid, whatever the majority prefers.

Reply:
With pinning to a specific version we could document that as the supported version and moving on to a later version could be performed in a more controlled fashion.
With using latest we might have surprise breaks even without changing code in this project. This wouldn't be that easy to investigate either. Also merging code would be blocked as long as we don't fix the incompatibility with a new Ubuntu version even though we could still deliver code using the release before.

I'm +1 to pin this to a specific version.

Reply:
Do we want to pin macOS too?

Reply:
It also depends how often these images change versions. If it's once every 2 years, it's less contentious to use "latest" than if GH bumps their versions every month.

Reply:
`ubuntu-latest` is about to be migrated to `24.04` and it was migrated to `22.04` from `20.04` around 2 years ago, see: https://github.com/actions/runner-images/commit/9c544f47011f3fb7f13ab122927f15939d950b49
`macos-latest` has been having a higher rotation, around once a year (it went from macos-11, to macos-12 and then to macos-14)

Reply:
I've pinned them to ubuntu-24.04 and macos-14